contract_manager/src/contracts/sui.ts

```diff
@@ -185,18 +173,56 @@ export class SuiContract extends Contract {
   async executeMigrateInstruction(vaa: Buffer, keypair: Ed25519Keypair) {
     const tx = new TransactionBlock();
     const packageId = await this.getPythPackageId();
-    let decreeReceipt = await this.getVaaDecreeReceipt(tx, packageId, vaa);
+    let verificationReceipt = await this.getVaaVerificationReceipt(
+      tx,
+      packageId,
+      vaa
+    );
 
     tx.moveCall({
       target: `${packageId}::migrate::migrate`,
-      arguments: [tx.object(this.stateId), decreeReceipt],
+      arguments: [tx.object(this.stateId), verificationReceipt],
     });
 
     return this.executeTransaction(tx, keypair);
   }
 
   async executeUpdatePriceFeed(senderPrivateKey: string, vaas: Buffer[]) {
-    throw new Error("Not implemented");
+    throw new Error("Use executeUpdatePriceFeedWithFeeds instead");
```

Comment:
I understand the complication here but it's not great to break the abstraction. Can you add some comments on why it's difficult to do and how we should make it work if we want to in the future?